### PR TITLE
Online learning

### PR DESCRIPTION
**Pull request by atihemati**
Hi @Mathias157,

I’ve updated the online-learning workflow:

- Exposed hyperparameters in the CLI: `--pretrain-epochs`, `--update-epochs`, `--days`, `--n-scenarios`, `--latent-dim`, `--seed`, `--batch-size`, `--learning-rate`.
- Threaded the same params into `pretrain()` / `train()` and the `ScenarioGenerator`.
- Added structured logging (streamed continuously) under `src/Logs/`.
- Updated `src/Jobs/online_learning.sh` to pass parameters directly.

Files changed:
- src/Workflow/Functions/neural_network.py
- src/Workflow/Functions/online_learning.py
- src/Jobs/online_learning.sh

Example run:
pixi run python Workflow/Functions/online_learning.py DO_D4W4 \
  --pretrain-epochs 50 \
  --update-epochs 100 \
  --days 1 \
  --n-scenarios 2 \
  --latent-dim 64 \
  --seed 0 \
  --batch-size 256 \
  --learning-rate 0.0005

I may later also test a richer ScenarioGenerator (deeper/residual/attention/variational) and training tweaks (schedulers, warmup, grad-clip.

Thanks!

### CODE REVIEW

**Comment by Mathias157** [comment]
Great! The master is very old, so I'll start merging my previous branching into it (before merging this one) so we can collaborate more easily
